scripts/calculate_percentile_offline.py

```diff
@@ -39,13 +39,24 @@
 
 # the algorithm
 def main():
+    stations_json_to_dict()
     parse_weather_dates()
     remove_data_outside_date_range()
     remove_data_outside_fire_season()
     sort_by_weather_station()
     calculate_percentile_per_station()
     write_output_to_json()
 
+def stations_json_to_dict():
+    with open('../data/weather_stations.json') as fileHandle:
+        weather_stations = json.loads(fileHandle.read())['weather_stations']
+    fileHandle.close()
+    global weather_stations_dict
```

Comment:
This step is also not needed, you could just read into a global weather_stations, and iterate through it later.